src/Pact/Types/Term.hs

```diff
@@ -608,41 +754,83 @@ data Term n =
     , _tMeta :: !Meta
     , _tInfo :: !Info
     }
-    deriving (Functor,Foldable,Traversable,Eq)
-
-instance Show n => Show (Term n) where
-    show TModule {..} =
-      "(TModule " ++ show _tModuleDef ++ " " ++ show (unscope _tModuleBody) ++ ")"
-    show (TList bs _ _) = "[" ++ unwords (map show bs) ++ "]"
-    show TDef {..} = show _tDef
-    show TNative {..} =
-      "(TNative " ++ asString' _tNativeName ++ " " ++ showFunTypes _tFunTypes ++ " " ++ unpack _tNativeDocs ++ ")"
-    show TConst {..} =
-      "(TConst " ++ asString' _tModule ++ "." ++ show _tConstArg ++ " " ++ show _tMeta ++ ")"
-    show (TApp a _) = show a
-    show (TVar n _) = "(TVar " ++ show n ++ ")"
-    show (TBinding bs b c _) = "(TBinding " ++ show bs ++ " " ++ show (unscope b) ++ " " ++ show c ++ ")"
-    show (TObject bs _ _) =
-      "{" ++ intercalate ", " (map (\(a,b) -> show a ++ ": " ++ show b) bs) ++ "}"
-    show (TLiteral l _) = show l
-    show (TGuard k _) = show k
-    show (TUse u _) = show u
-    show (TValue v _) = BSL.toString $ encode v
-    show (TStep ent e r _) =
-      "(TStep " ++ show ent ++ " " ++ show e ++ maybeDelim " " r ++ ")"
-    show TSchema {..} =
-      "(TSchema " ++ asString' _tModule ++ "." ++ asString' _tSchemaName ++ " " ++
-      show _tFields ++ " " ++ show _tMeta ++ ")"
-    show TTable {..} =
-      "(TTable " ++ asString' _tModule ++ "." ++ asString' _tTableName ++ ":" ++ show _tTableType
-      ++ " " ++ show _tMeta ++ ")"
+    deriving (Functor,Foldable,Traversable,Eq,Show)
+
+-- | Newtype to avoid introducing an orphan instance for @Pretty Var@. Used
+-- just for pretty printing (@prettyUnscope@).
+newtype PrettyVar a b = PrettyVar (Var a b)
+
+instance (Pretty a, Pretty b) => Pretty (PrettyVar a b) where
+  pretty (PrettyVar (B b)) = pretty b
+  pretty (PrettyVar (F a)) = pretty a
+
+prettyUnscope
+  :: (Functor f, Pretty (f (PrettyVar b (f a))))
+  => Scope b f a -> Doc
+prettyUnscope = pretty . fmap PrettyVar . unscope
+
+instance Pretty n => Pretty (Term n) where
+  pretty = \case
+    TModule{..} -> parensSep
+      [ "TModule", pretty _tModuleDef, prettyUnscope _tModuleBody ]
+    TList{..} -> prettyList _tList
+    TDef{..} -> pretty _tDef
+    TNative{..} -> dullgreen ("native function `" <> pretty (asString' _tNativeName) <> "`")
+      <> nest 2 (
+         line
+      <> line <> pretty _tNativeDocs
+      <> line
+      <> line <> dullgreen "Examples:"
+      <> line <> align (vsep (pretty <$> _tNativeExamples))
+      <> line
+      <> line <> dullgreen "Type:"
+      <> line <> align (vsep (pretty <$> toList _tFunTypes))
```

Comment:
these funtypes are really hard to read. I think we'll need to iterate on a better format than 

`x:<a [string....] y:<a [foo....] -> ...`

It's not particularly legible, and this gives us the chance to change it up for everyone's sake.

Reply:
:100: I was hoping to punt this to a later review though, since I'm anticipating some debate

Reply:
Better discussed in the `Type` module